src/main/groovy/com/anotherchrisberry/spock/extensions/retry/RetrySpecExtension.groovy

```diff
@@ -44,7 +46,25 @@ class RetrySpecExtension extends AbstractAnnotationDrivenExtension<RetryOnFailur
     }
     double getDelaySeconds(RetryOnFailure retries) {
         String defaultDelay = Double.toString(retries.delaySeconds())
-        return Double.parseDouble(System.getProperty("spck-retry.delaySeconds", defaultDelay))
+        return Double.parseDouble(System.getProperty("spock-retry.delaySeconds", defaultDelay))
+    }
+
+    Predicate<Throwable> getPredicate(final RetryOnFailure retries) {
+        final Predicate<Throwable> predOnly = Optional.ofNullable(retries.only())
+                .map{l -> Arrays.asList(l)}
+                .filter{l -> ! l.isEmpty()}
+                .map{l -> new HashSet<Throwable>(l)}
+                .map{c -> {t -> hasMatchingClass(t, c)}}
+                .orElseGet { -> {t -> true} }
+
+        final Predicate<Throwable> predExcludes = Optional.ofNullable(retries.exclude())
+                .map{l -> Arrays.asList(l)}
+                .filter{l -> ! l.isEmpty()}
+                .map{l -> new HashSet<Throwable>(l)}
+                .map{c -> {t -> ! hasMatchingClass(t, c)}}
+                .orElseGet { -> {t -> true} }
```

Comment:
Can this be simplified to:
```suggestion
        final Predicate<Throwable> predExcludes = Optional.ofNullable(retries.exclude())
                .map{l -> new HashSet<Throwable>(Arrays.asList(l))}
                .map{c -> {t -> !hasMatchingClass(t, c)}}
                .orElseGet { -> {t -> true} }
```
I don't think you necessarily need to `filter`, do you?

Reply:
Yes. Method has matchingClass returns a relevant result only on non empty collections.
To simplify it i removed the HashSet stuff which is unnecessary.